### PR TITLE
GH-102821: threading.Event.wait - Improve logic

### PR DESCRIPTION
**Pull request by mementum**
https://github.com/python/cpython/issues/102821

* Issue: gh-102821

### CODE REVIEW

**Comment by Eclips4** [comment]
It seems like only a cosmetic changes, and doesn't affect on performance at all ( or it does, but very weakly ).

By other hand, this PR carries more changes than expected. You're aplied PEP8 to all module, but it should be a separate issue/PR.

**Comment by mementum** [comment]
> It looks like only a cosmetic changes, and doesn't affect on performance at all ( or it does, but very weakly ).

Indeed, I simply thought that by making things clearer and avoiding unneeded variable assignment, things could be "slightly" better.

I don't have a feeling with regards to how such changes are welcome, needed or sought after in CPython (I have read some other modules and would have many other proposals for simplification, modernization, call it x, without changing functionality)

> By other hand, this PR carries more changes than expected. You're aplied PEP8 to all module, but it should be a separate issue/PR.

Sorry about that, the PEP-8 should have gone into a different branch. I was distracted by the "skip news" notification. Being this my first attempt at a contribution to CPython, it disrupted my attempt at a clean workflow.

I will separate both commits for clarity.

**Comment by Eclips4** [comment]
Don't worry! =)
Contributions to CPython always are welcome.
Altough, I don't know what core devs think about refactoring stdlib.

Just FYI, stdlib of CPython have a certain amount of modules/packages which is not following PEP8.
We can't say that this is a problem, but, if anyone decide to solve this, he must use a tool designed for this.

**Review by corona10 — CHANGES_REQUESTED** [review]
Please revert the code style changes, it makes difficult to track the history of modifications

File: Misc/NEWS.d/next/Library/2023-03-19-08-01-48.gh-issue-102821.yBn1x-.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve threading.Event.wait logic
```
Comment (corona10) [inline Misc/NEWS.d/next/Library/2023-03-19-08-01-48.gh-issue-102821.yBn1x-.rst]: No NEWS.d is needed for this kind of change.

**Pull request closed without merging**